notebooks/WFC3/ir_ima_visualization/IR_IMA_Visualization_with_an_Example_of_Time_Variable_Background.ipynb

```diff
@@ -96,7 +96,6 @@
     "- *matplotlib.pyplot* for plotting data\n",
```

Comment:
<p>Thank you for the updates you made to the notebook. I enjoyed reading through it and learned a lot from the approach. Here are a few technical suggestions for further enhancement:</p><ul><li>I&nbsp;recommend using context managers when opening FITS files.</li><li>I noticed that the&nbsp;ax.set_title&nbsp;and&nbsp;ax.imshow&nbsp;calls are repeated in both the&nbsp;if&nbsp;and&nbsp;else&nbsp;blocks. This repetition can be reduced by moving the common code outside of the conditional blocks and only changing what differs between them, such as the&nbsp;vmin&nbsp;and&nbsp;vmax&nbsp;values for&nbsp;ax.imshow. This will make the code more concise and easier to maintain.</li></ul><p> </p><pre class="ql-syntax" spellcheck="false">fig = plt.figure(figsize=(30, 30))
rows = 2
columns = 2


files = [ima_scattered, flt_scattered, ima_nominal, flt_nominal]
subplot_titles = ['scattered', 'nominal']


for i, file in enumerate(files):
&nbsp; &nbsp; path, filename = os.path.split(file)


&nbsp; &nbsp; with fits.open(file) as image:
&nbsp; &nbsp; &nbsp; &nbsp; ax = fig.add_subplot(rows, columns, i+1)
&nbsp; &nbsp; &nbsp; &nbsp; title = f'{filename}, {subplot_titles[i//2]}'
&nbsp; &nbsp; &nbsp; &nbsp; ax.set_title(title, fontsize=20)


&nbsp; &nbsp; &nbsp; &nbsp; vmin, vmax = (0.25, 1.7) if i &lt; 2 else (0.5, 1.2)
&nbsp; &nbsp; &nbsp; &nbsp; im = ax.imshow(image["SCI", 1].data, origin='lower', cmap='Greys_r', vmin=vmin, vmax=vmax)
&nbsp; &nbsp; &nbsp; &nbsp; plt.colorbar(im, ax=ax)


plt.subplots_adjust(bottom=0.2, right=0.5, top=0.5)
plt.rc('xtick', labelsize=10)
plt.rc('ytick', labelsize=10)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/hst_notebooks/pull/106/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>